File: Sample0.7.2/helloworld/src/main/python/UresujiConsumer.py
#! /usr/local/Python-2.7.5/bin/python
# coding:utf-8
from kazoo.client import KazooClient
from samsa.cluster import Cluster
import json
import time
import redis
zookeeper = KazooClient(hosts='localhost:2181')  #...(1)
zookeeper.start()                                #...(2)
cluster = Cluster(zookeeper)                     #...(3)
topic = cluster.topics['topic1']                 #...(4)
consumer = topic.subscribe('group-name1')        #...(5)
r = redis.Redis(host='localhost',port=6379,db=0) #...(6)
for message in consumer:                         #...(7)
    jmes=json.loads(message)                     #...(8)
    shohinlist = jmes['shohinlist']
    for shohin in shohinlist:                    #...(9)
        name = shohin['name']
        count = shohin['count']
        r.zincrby('ranking', name, count)        #...(10)

File: Sample0.7.2/helloworld/src/main/ruby/UresujiConsumer.rb
#! /usr/local/ruby1.9/bin/ruby
# -*- encoding: utf-8 -*-
require 'kafka'
require 'json'
require 'redis'

consumer = Kafka::Consumer.new({            # ...(1)
  :host => 'localhost',
  :port => 9092,
  :topic =>'topic1',
  :partition => 0,
  :offset => 0
})

redis = Redis.new({                         # ...(2)
  :host => 'localhost',
  :port => 6379
})
redis.ping

p "Consumer started."
consumer.loop do |messages|                 # ...(3)
  messages.each do |message|                # ...(4)
    jmes = JSON.parse(message.payload)      # ...(5)
#    puts message.payload
    time = jmes['time']
    denpyoId = jmes['denpyo-id']
    shohinlist = jmes['shohinlist']
    shohinlist.each do |shohin|             # ...(6)
      name = shohin['name']
      count = shohin['count']
      redis.zincrby "ranking", count, name  # ...(7)
    end
  end
end
